src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -95,13 +95,25 @@ public async Task StartContainersAsync(IExecutionContext executionContext, objec
             Version requiredDockerVersion = new Version(17, 12);
 #endif
 
-            if (dockerVersion.ServerVersion < requiredDockerVersion)
+            var serverType = executionContext.Variables.Get(Constants.Variables.System.ServerType);
+            var agentOS = executionContext.Variables.Get(Constants.Variables.Agent.OS);
+
+            var checkDockerVersion = !(string.Equals(serverType, "Hosted", StringComparison.OrdinalIgnoreCase) &&  string.Equals(agentOS, "Linux", StringComparison.OrdinalIgnoreCase));
```

Comment:
`serverType` == `Hosted` means VSTS, not On-prem TFS.
So, your logic will apply to customer who use their private linux agent configure against VSTS as well.